src/core/qgsgdalutils.cpp

```diff
@@ -97,61 +97,46 @@ gdal::dataset_unique_ptr QgsGdalUtils::createSingleBandTiffDataset( QString file
   return hDstDS;
 }
 
-gdal::dataset_unique_ptr QgsGdalUtils::imageToMemoryDataset( const QImage &image, QgsRectangle extent, const QgsCoordinateReferenceSystem &crs )
+gdal::dataset_unique_ptr QgsGdalUtils::imageToMemoryDataset( const QImage &image )
```

Comment:
I haven't check at the comments of the method, but there should be a mention that the exact QImage objet passed to this method (or at the very list its pixel array) should be kept alive (and not modified, if QImage is mutable) while the GDALDataset is in use, otherwise it will point to garbage

Reply:
Good point. I'll probably make all these private anyway...